relayer/chain/substrate/listener.go

```diff
@@ -149,6 +161,122 @@ func (li *Listener) pollBlocks(ctx context.Context) error {
 	}
 }
 
+func (li *Listener) subBeefyJustifications(ctx context.Context) error {
+	ch := make(chan interface{})
+
+	sub, err := li.relayconn.api.Client.Subscribe(context.Background(), "beefy", "subscribeJustifications", "unsubscribeJustifications", "justifications", ch)
+	if err != nil {
+		panic(err)
+	}
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return li.onDone(ctx)
+		case msg := <-ch:
+
+			signedCommitment := &store.SignedCommitment{}
+			err := types.DecodeFromHexString(msg.(string), signedCommitment)
+			if err != nil {
+				li.log.WithError(err).Error("Failed to decode BEEFY commitment messages")
+			}
+
+			li.log.Info("Witnessed a new BEEFY commitment: \n", msg.(string))
+			if len(signedCommitment.Signatures) == 0 {
+				li.log.Info("BEEFY commitment has no signatures, skipping...")
+				continue
+			}
+
+			signedCommitmentBytes, err := json.Marshal(signedCommitment)
+			if err != nil {
+				li.log.WithError(err).Error("Failed to marshal signed commitment:", signedCommitment)
+				continue
+			}
+
+			// TODO:
+			// beefyAuthorities, err := li.getBeefyAuthorities(uint64(signedCommitment.Commitment.BlockNumber))
+			// if err != nil {
+			// 	li.log.WithError(err).Error("Failed to get Beefy authorities from on-chain storage")
+			// }
+
+			beefyAuthorities := []common.Address{
+				common.HexToAddress("0xE04CC55ebEE1cBCE552f250e85c57B70B2E2625b"),
+				common.HexToAddress("0x25451A4de12dcCc2D166922fA938E900fCc4ED24"),
+			}
+
+			beefyAuthoritiesBytes, err := json.Marshal(beefyAuthorities)
+			if err != nil {
+				li.log.WithError(err).Error("Failed to marshal BEEFY authorities:", beefyAuthorities)
+				continue
+			}
+
+			item := store.BeefyItem{
+				ValidatorAddresses: beefyAuthoritiesBytes,
+				SignedCommitment:   signedCommitmentBytes,
+				Status:             store.CommitmentWitnessed,
+			}
+
+			li.log.Info("1: Writing BEEFY item to database with status 'WitnessedCommitment'")
+			cmd := store.NewDatabaseCmd(&item, store.Create, nil)
+			li.beefyMessages <- cmd
+		}
+	}
+}
+
+type Authorities = [][33]uint8
```

Comment:
move to another file plz

Reply:
Updated in https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/179de7d5ab82396d4d1a473b7b2a1ff4f5e9c302